indy_node/test/auth_rule/test_auth_rules_transaction.py

```diff
@@ -0,0 +1,134 @@
+import pytest
+
+from indy_common.authorize.auth_actions import ADD_PREFIX, EDIT_PREFIX
+from indy_common.authorize.auth_constraints import ROLE, SIG_COUNT
+from indy_common.constants import AUTH_ACTION, OLD_VALUE, CONSTRAINT, NEW_VALUE, NYM, TRUST_ANCHOR
+from indy_node.test.auth_rule.helper import generate_constraint_entity, generate_constraint_list, \
+    sdk_send_and_check_auth_rules_request, generate_auth_rule_operation, sdk_send_and_check_auth_rules_request, \
+    sdk_get_auth_rule_request, generate_auth_rule
+from plenum.common.constants import TRUSTEE, STEWARD, DATA
+from plenum.common.exceptions import RequestRejectedException, \
+    RequestNackedException
+
+RESULT = "result"
+
+
+def test_auth_rules_transaction_without_changes(looper,
+                                                sdk_wallet_trustee,
+                                                sdk_pool_handle):
+    before_resp = sdk_get_auth_rule_request(looper,
+                                            sdk_wallet_trustee,
+                                            sdk_pool_handle)
+    sdk_send_and_check_auth_rules_request(looper,
+                                          sdk_wallet_trustee,
+                                          sdk_pool_handle,
+                                          rules=before_resp[0][1][RESULT][DATA])
+    after_resp = sdk_get_auth_rule_request(looper,
+                                           sdk_wallet_trustee,
+                                           sdk_pool_handle)
+    assert before_resp[0][1]["result"][DATA] == after_resp[0][1]["result"][DATA]
+
+
+def test_auth_rules_transaction(looper,
+                                sdk_wallet_trustee,
+                                sdk_pool_handle):
+    rule = generate_auth_rule()
+    key = dict(rule)
+    key.pop(CONSTRAINT)
```

Comment:
Why do we remove `CONSTRAINT` here?

Reply:
Because it's the variable for getting a constraint by the action key.